### PR TITLE
misc: handle connection errors even before the timeout

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Should fix travis and give us better error messages when Chrome crashes unexpectedly.

Previously we only handled connection errors after we reached our timeout when we should be handling them as soon as we try to connect in case Chrome crashes unexpectedly.

**Related Issues/PRs**
#7534

### CODE REVIEW

**Comment by patrickhulce** [comment]
So the errors smoke test has been fixed :)

but the NO_TRACING_STARTED is still very in our face :(

File: lighthouse-core/gather/connections/cri.js [hunk 1]
```diff
@@ -108,16 +108,17 @@ class CriConnection extends Connection {
         });
       });
 
-      request.setTimeout(CONNECT_TIMEOUT, () => {
-        request.abort();
+      let ignoreECONNRESET = false;
```
Comment (brendankenny) [inline lighthouse-core/gather/connections/cri.js]: can you add a comment explaining this

Comment (patrickhulce) [reply lighthouse-core/gather/connections/cri.js]: as in move up the comment that's where we set it or a new one duplicating? :)

**Comment by patrickhulce** [comment]
> so what changed with 74 exactly?

Chrome's shutdown procedure is clearly different though I'm not sure what exactly. If I had to guess I would say it's related to network service since that's dramatically changed how Chrome shuts down (usually takes 4-6s now locally).

File: lighthouse-core/gather/connections/cri.js (hunk 1 above)
Comment (brendankenny) [reply lighthouse-core/gather/connections/cri.js]: duplicate-ish?

Comment (patrickhulce) [reply lighthouse-core/gather/connections/cri.js]: refactored so it's not necessary instead

Comment (brendankenny) [reply lighthouse-core/gather/connections/cri.js]: love it

**Pull request merged**